modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -306,13 +324,36 @@ public CommerceOrder addCommerceOrder(
 		commerceOrder.setCompanyId(user.getCompanyId());
 		commerceOrder.setUserId(userId);
 		commerceOrder.setUserName(user.getFullName());
-		commerceOrder.setBillingAddressId(billingAddressId);
+
+		if (billingAddressId > 0) {
+			commerceOrder.setBillingAddressId(billingAddressId);
+		}
+		else {
+			_setAccountDefaultBillingAddress(commerceChannel, commerceOrder);
+		}
+
 		commerceOrder.setCommerceAccountId(commerceAccountId);
 		commerceOrder.setCommerceCurrencyCode(commerceCurrencyCode);
 		commerceOrder.setCommerceOrderTypeId(commerceOrderTypeId);
 		commerceOrder.setCommerceShippingMethodId(commerceShippingMethodId);
-		commerceOrder.setShippingAddressId(shippingAddressId);
-		commerceOrder.setCommercePaymentMethodKey(commercePaymentMethodKey);
+
+		if (shippingAddressId > 0) {
+			commerceOrder.setShippingAddressId(shippingAddressId);
+		}
+		else {
+			_setAccountDefaultShippingAddress(commerceChannel, commerceOrder);
+		}
+
+		if (commercePaymentMethodKey != null) {
+			commerceOrder.setCommercePaymentMethodKey(commercePaymentMethodKey);
+		}
+		else {
+			_setAccountDefaultPaymentIntegrationKey(
+				commerceChannel, commerceOrder);
+		}
+
+		_setAccountDefaultPaymentTerm(commerceChannel, commerceOrder, user);
+
```

Comment:
Couple questions and observations here:

1. Previous PR had `_setAccountDefaultDeliveryTerm` as well? Was this intentionally removed?